src/socket/udp.rs

```diff
@@ -94,6 +97,30 @@ impl<'a, 'b> UdpSocket<'a, 'b> {
         self.endpoint
     }
 
+    /// Return the time-to-live (IPv4) or hop limit (IPv6) value used in outgoing packets.
+    ///
+    /// See also the [set_ttl](#method.set_ttl) method
+    pub fn ttl(&self) -> Option<u8> {
+        self.ttl
+    }
+
+    /// Set the time-to-live (IPv4) or hop limit (IPv6) value used in outgoing packets.
+    ///
+    ///  * A socket without an explicitly set TTL value uses the default [INNA recommended]
+    ///    value (`64`).
+    ///  * A TTL value of 0 is invalid and will be ignored. See [RFC 1122 § 3.2.1.7].
```

Comment:
Added documentation about the added logic and updated documentation about the default value.